AAUnlimited/Functions/AAPlay/Subs.cpp

```diff
@@ -1,33 +1,147 @@
 #include "StdAfx.h"
 
 #include <list>
+#include <regex>
 
 namespace Subtitles {
 
-	std::wstring text;
-	std::list<std::wstring> lines;
-	RECT rect = { 0, 40, 1024, 256 };
-	D3DCOLOR color = D3DCOLOR_RGBA(255, 255, 255, 255);
+	std::list<std::tuple< std::wstring, int >> lines;
+	const int fontLayersNum = 9;
+	RECT rect[fontLayersNum] = {
+		{ -1, 39, 1024, 256 }, { 1, 39, 1024, 256 }, 
+		{ 1, 41, 1024, 256 }, { -1, 41, 1024, 256 },
+		{ -1, 40, 1024, 256 },{ 1, 40, 1024, 256 },
+		{ 0, 39, 1024, 256 },{ 0, 41, 1024, 256 },
+		{ 0, 40, 1024, 256 } 
+	};
+	// 0 - outline, 1 - female, 2 - male
+	D3DCOLOR colors[3] = { D3DCOLOR_RGBA(0, 0, 0, 255), 
```

Comment:
This is a great idea but all the lines on display were on a single string for a reason, DrawText is an expensive function to call every frame so it just got called once before. You should go back to the previous approach and have a single color for subtitles text.

Reply:
answered in main comment